src/quotiauth.js

```diff
@@ -139,6 +151,13 @@ class QuotiAuth {
           if (!permissionsResult) {
             throw new Error('Insufficient permissions or user is null')
           }
+        } else if (permissions && permissions.permissionsToValidate && permissions.permissionsToValidate.length) {
```

Comment:
Recomendo reestruturar o código para evitar repetição.

Algo desse tipo (calcular o argumento que tu vai passa pro this.validateSomePermissionCluster antes de chamar ele):

`const permissionsToValidate = permissions?.permissionsToValidate || permissions`